### PR TITLE
ci: fix flaky e2e tests

### PR DESCRIPTION
**Pull request by zirain**
- improve local ratelimit tests, remove check of `RatelimitRemainingHeaderName`.
  - That could also be a edge case, as the current token is not deducted one by one as an integer, but is a floating-point number calculated based on the fill rate, initialization time, and the maximum fill number of three counts. However, when writing the entire remaining header, it is necessary to forcibly convert the floating-point number to an integer number, which will result in floating on the display.
- wait fro BTLS policy before starting HTTPRouteMixedProtocols test, also make sure the backend is ready before verification.
- Adopt `DefaultPollInterval` to reduce the noisy logs in CI pipeline.
- Adjust `RequiredConsecutiveSuccesses`.

### CODE REVIEW

File: test/e2e/tests/local_ratelimit.go [hunk 1]
```diff
@@ -189,7 +189,7 @@ func runSpecificUserRateLimitTest(t *testing.T, suite *suite.ConformanceTestSuit
 	if !disableHeader {
 		okResponse.Response.Headers = map[string]string{
 			RatelimitLimitHeaderName:     "3",
-			RatelimitRemainingHeaderName: "1",
```
Comment (zirain) [inline test/e2e/tests/local_ratelimit.go:192]: sometime the bucket decrease from 2 to 0 directly:
```
    local_ratelimit.go:443: 2026-03-02T04:06:29.437753617Z: Making GET request to host  via http://172.18.0.203/ratelimit-path-match/foo
    http.go:285: 2026-03-02T04:06:29.441003173Z: Response expectation failed for request: {URL: {Scheme:http Opaque: User: Host:172.18.0.203 Path:/ratelimit-path-match/foo RawPath: OmitHost:false ForceQuery:false RawQuery: Fragment: RawFragment:}, Host: , Protocol: HTTP, Method: GET, Headers: map[X-Echo-Set-Header:[]], UnfollowRedirect: false, ServerName: , ServerCertificate: <truncated>, ClientCertificate: <truncated>, ClientCertificateKey: <truncated>}  not ready yet: expected x-ratelimit-remaining header to be set to 1, got 2 (after 1.463µs)
    http.go:285: 2026-03-02T04:06:30.446341074Z: Response expectation failed for request: {URL: {Scheme:http Opaque: User: Host:172.18.0.203 Path:/ratelimit-path-match/foo RawPath: OmitHost:false ForceQuery:false RawQuery: Fragment: RawFragment:}, Host: , Protocol: HTTP, Method: GET, Headers: map[X-Echo-Set-Header:[]], UnfollowRedirect: false, ServerName: , ServerCertificate: <truncated>, ClientCertificate: <truncated>, ClientCertificateKey: <truncated>}  not ready yet: expected x-ratelimit-remaining header to be set to 1, got 0 (after 1.006753669s)
```

Comment (zirain) [reply test/e2e/tests/local_ratelimit.go:192]: That could also be a edge case, as the current token is not deducted one by one as an integer, but is a floating-point number calculated based on the fill rate, initialization time, and the maximum fill number of three counts. However, when writing the entire remaining header, it is necessary to forcibly convert the floating-point number to an integer number, which will result in floating on the display.
